### PR TITLE
[FEAT] Demo time/frequency monitoring: enable/disable overlays and paths

### LINKED ISSUE #187: [FEAT] Demo time/frequency monitoring: highlight paths
Improve the demo created with #179

Add a way to highlight paths
- time: longest/fastest execution
- frequency: less/most used

We can use something like we have put in the CSS style example.
Alternatively, we could hide the other paths or make them less visible.

highlight path

### Tasks

- [x] display hightlight by default -  #198
- [x] allow to enable/disable path highlighting - #221

### LINKED ISSUE #219: [FEAT] Demo time/frequency monitoring: allow to enable/disable overlays
Improve the demo updated with #187

Allow to enable/disable overlays. We currently don't have an API to do this. Workaround: for each bpmn element, [remove overlays](https://github.com/process-analytics/bpmn-visualization-js/issues/1168)

### PR DESCRIPTION
**Pull request by tbouffard**
closes #187 
closes #219

**Live environment of examples**

https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/187-monitoring_demo_toggle_highlight_paths/examples/index.html

### CODE REVIEW

**Comment by tbouffard** [comment]
### 1st implementation

1st_implementation

**Comment by csouchet** [comment]
I tried to separate the example panel from the panel button.

### Grey background
image

### Dark background
image

### Primary background
image

### Secondary background
image

### Success background
image

### Warning background
image

### Error background
image

**Comment by tbouffard** [comment]
Using a background is a nice idea, my favorites are in order
- dark 
- secondary

As the background border, we could also do it for the bpmn-container border. Especially with the dark baground as in this case, we don't see the bpmn-container border

**Comment by csouchet** [comment]
Another idea (not good position, or color) 
image

**Comment by NathalieC** [comment]
1. Layout
To me, this should look like a group of 2 sliders, one on top of the other or the first on the left, and the second just to the right of the first one.
I hope you can find some. Here is an example of a multi-position slider:
image

Each slider would have 3 positions:
- First slider: "Average duration" (or runtime) - "Regular view" - "Number of occurrences" (or frequency)
Its color is deep purple, as it is now.
By default, the slider in on "Regular view", and the second slider is disabled.
When the user takes the first slider to either position 1 or 3, the second slider becomes enabled:
- Second slider: "Paths" - "Overlays" - "Both"
Its color is light purple.

Each **combination** of sliders options should give a specific name to the graph.

This will avoid the situation where neither button is pressed but the view is still "time" or "frequency".

2. Utility
I do not get the "Path" option for "Time": can we calculate the duration of each transition? Not all transitions seem to have an overlay, why?
If the duration is about the paths, then the graph name should be "average runtime for each path", and the user should be able to choose the path to look at, have it highlighted, and have all others be grayed out.

3. Meaning of the color / thickness code
"Time" display: when "Paths" is not active (second slider on "Overlays"), I would expect the transitions to be gray and thin, and when "Paths" is "active", I would expect them to be black and with different thicknesses. Here they are gray with different thicknesses or black and thin. As a result, I do not really understand when "Path" is active compared to the "Regular view". => Regular view should have light gray transitions, or the highlight of the path should always be more visible than the regular transitions.

4. Visual feedback
If you do not want the second slider but want to keep the buttons, when "Overlays" and "Paths" are toggled on, the buttons should look "pressed", and when "toggled off", the buttons should look "released" ; so at any time, we can see by looking at the buttons what is toggled on and off.

5. I like the color of the panel around the pool ; either dark or beige is fine I would say.
The sliders can stay on top, and the title of the graph be centered (left-right).

File: demo/monitoring-all-process-instances/js/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+document.getElementById('choose-data-panel').onclick = () => {
+    const bpmnVisualization = currentDiagram === 'time' ? timeBpmnVisualization : frequencyBpmnVisualization;
+    const data = currentDiagram === 'time' ? timeData : frequencyData;
+    switchData(bpmnVisualization, data);
+}
+
+
+
+
```
Comment (tbouffard) [inline demo/monitoring-all-process-instances/js/index.js:110]: extra lines
```suggestion
```

**Review by tbouffard — COMMENTED** [review]
This is a first workgin implementation, the design will be improved in incoming Pull Requests.
Current design is as in the following.

image

**Review by csouchet — APPROVED** [review]

**Pull request merged**